dot-net-sdk/store/ConfigurationStore.cs

```diff
@@ -8,83 +8,48 @@ namespace eppo_sdk.store;
 
 public class ConfigurationStore : IConfigurationStore
 {
-    private readonly MemoryCache _experimentConfigurationCache;
+    private readonly MemoryCache _flagConfigurationCache;
     private readonly MemoryCache _banditModelCache;
     private readonly ConfigurationRequester _requester;
-    private static ConfigurationStore? _instance;
 
-    public ConfigurationStore(ConfigurationRequester requester, MemoryCache flagConfigurationCache, MemoryCache banditModelCache)
+    public ConfigurationStore(ConfigurationRequester requester,
+                              MemoryCache flagConfigurationCache,
+                              MemoryCache banditModelCache)
     {
         _requester = requester;
-        _experimentConfigurationCache = flagConfigurationCache;
+        _flagConfigurationCache = flagConfigurationCache;
         _banditModelCache = banditModelCache;
     }
 
-    public static ConfigurationStore GetInstance(MemoryCache flagConfigurationCache, MemoryCache banditModelCache,
-        ConfigurationRequester requester)
-    {
-        if (_instance == null)
-        {
-            _instance = new ConfigurationStore(requester, flagConfigurationCache, banditModelCache);
-        }
-        else
-        {
-            _instance._experimentConfigurationCache.Clear();
-        }
-
-        return _instance;
-    }
-
     public void SetExperimentConfiguration(string key, Flag experimentConfiguration)
     {
-        _experimentConfigurationCache.Set(key, experimentConfiguration, new MemoryCacheEntryOptions().SetSize(1));
+        _flagConfigurationCache.Set(key, experimentConfiguration, new MemoryCacheEntryOptions().SetSize(1));
     }
 
     public void SetBanditModel(Bandit banditModel)
     {
-        _banditModelCache.Set(banditModel.BanditKey, banditModel, new MemoryCacheEntryOptions().SetSize(1));
-    }
-
-    public Flag? GetExperimentConfiguration(string key)
-    {
-        try
-        {
-            if (_experimentConfigurationCache.TryGetValue(key, out Flag? result))
-            {
-                return result;
-            }
-        }
-        catch (Exception)
-        {
-            throw new ExperimentConfigurationNotFound($"[Eppo SDK] Experiment configuration for key: {key} not found.");
-        }
-
-        return null;
+        _banditModelCache.Set(
+            banditModel.BanditKey,
+            banditModel,
+            new MemoryCacheEntryOptions().SetSize(1));
     }
 
+    public bool TryGetFlag(string key, out Flag? result) => _flagConfigurationCache.TryGetValue(key, out result);
```

Comment:
`TryGetValue` is the idiomatic pattern in .NET for testing for presence + retrieval.